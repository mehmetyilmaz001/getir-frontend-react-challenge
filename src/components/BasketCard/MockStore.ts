import { Basket, BasketItem } from './../../types/Basket';
export const mockBasketItem: BasketItem = {
    added: 213213123,
    description: "",
    itemType: "",
    manufacturer: "",
    slug: "",
    tags: [],
    name: "test",
    price: 111,
    quantity: 1,
  };

export const mockBaskeCardStore: Basket = {
    items: [mockBasketItem],
    total: 2222
}