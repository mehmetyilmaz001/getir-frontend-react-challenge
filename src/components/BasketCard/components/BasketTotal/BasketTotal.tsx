import React from 'react';
import styled from 'styled-components';
import mainThememe from '../../../../style/Theme';
import FormattedLabel from '../../../common/FormattedLabel/FormattedLabel';
import { ReactComponent as BasketIcon } from '../../../../assets/basket-icon.svg';
import { useSelector } from 'react-redux';
import { Store } from '../../../../redux';
import { mq } from '../../../../style/Mixins';


interface BasketTotalProps {
    showBasketIcon: boolean;
    theme: 'light' | 'dark';
}
const Total = styled.div<{
    theme: Pick <BasketTotalProps, "theme">;
}>`
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    border: ${({theme}) => theme === 'light' ? '2px' : 0} solid ${mainThememe.palette.primary.main};
    color: ${({theme}) => theme === 'light' ? mainThememe.palette.primary.main : mainThememe.palette.primary.text};
    background-color: ${({theme}) => theme === 'light' ? mainThememe.palette.primary.text : mainThememe.palette.accent.main};
    font-size: 14px;
    font-weight: 600;
    gap: 13px;

    ${mq('mobile')} {
       flex-direction : column;
    }
`;
 
export const BasketTotal: React.FunctionComponent<BasketTotalProps> = ({showBasketIcon, theme}) => {
    const { total, items } = useSelector((s:Store) => s.basket);
    if(items.length === 0) {
        return <></>
    }
    return ( 
        <Total theme={theme}>
            {showBasketIcon && <BasketIcon />}
            <FormattedLabel price={total} />
        </Total>
     );
}

export default BasketTotal;