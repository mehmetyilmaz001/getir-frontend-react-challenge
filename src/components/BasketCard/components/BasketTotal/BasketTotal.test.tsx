import BasketTotal from "./BasketTotal";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import renderer from 'react-test-renderer';
import { mockBaskeCardStore } from "../../MockStore";


const mockStore = configureStore([]);

test("Basket total renders from redux store correctly", async () => {
  let store;
  let component: any;

    store = mockStore({
       basket: {
        ...mockBaskeCardStore
       }  
    });

    component = renderer.create(
        <Provider store={store}>
          <BasketTotal showBasketIcon={false} theme="light" />
        </Provider>
      );
 

  expect(component!.toJSON()).toMatchSnapshot();
});
