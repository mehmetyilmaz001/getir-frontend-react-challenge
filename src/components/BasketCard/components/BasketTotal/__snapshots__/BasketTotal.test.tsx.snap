// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basket total renders from redux store correctly 1`] = `
<div
  className="sc-bdvvtL eYgSuc"
>
  <span>
    â‚º2.222,00
  </span>
</div>
`;
